packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart.ts

```diff
@@ -143,7 +147,10 @@ export class ForceFunctionStartExecutor extends SfdxCommandletExecutor<string> {
     );
 
     execution.stdoutSubject.subscribe(data => {
-      if (data.toString().includes('Debugger running on port')) {
+      const matches = String(data).match(runtimeDetection);
+      if (matches && matches.length > 1) {
+        FunctionService.instance.updateFunction(functionDirPath, matches[1]);
+      } else if (data.toString().includes('Debugger running on port')) {
```

Comment:
why is here an else condition? Don't we want to close the notification irrespective of a language match or not.

Reply:
The language match occurs fairly early, whereas the "running" message is one of the last messages.

Reply:
Synced offline and now understand data is one line of stdout at a time. So, this is fine.